pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -59,6 +61,7 @@ type wlReconciler struct {
 	clusters          *clustersReconciler
 	origin            string
 	workerLostTimeout time.Duration
+	deletedWlCache    *utilmaps.SyncMap[string, *kueue.Workload]
```

Comment:
I'm wondering if this could leak if MultiKueueGC removes the workload in the meanwhile, when the synchronous delete could fail, leaking the entry. Should we also delete the entry when performing MultiKueueGC?

Reply:
The NotFound errors are ignored during the object deletion, other errors will trigger a new reconcile that will eventually get to remove the cache entry.
(I have extended a bit the unit tests)